string ip
string qr
string found